app/models/course.rb

```diff
@@ -71,7 +71,10 @@ def import_students(file, header_map, header_row_exists)
       next if row.values.all?(&:nil?) # skip empty rows
 
       # check if there is an existing student in the course or create a new one
-      student = roster_students.find_by(perm: row["student_id"]) || roster_students.new
+      student = roster_students.find_by(perm: row["student_id"]) ||
+      roster_students.find_by(email: row["email"]) ||
+      roster_students.new
```

Comment:
@wpeery what made it necessary to search by email out of curiosity?

If we are viewing email and student_id as unique constrained fields it might be a good idea to make sure these fields have unique constraints on them in the database schema -- this code appears to assume that they are indeed unique to any roster student so this is needed for the sake of database consistency :)